eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java

```diff
@@ -85,43 +95,106 @@ public EventHubClientBuilder scheduler(Scheduler scheduler) {
      * @param proxyConfiguration The proxy configuration to use.
      * @return The updated EventHubClientBuilder object.
      */
-    public EventHubClientBuilder proxy(ProxyConfiguration proxyConfiguration) {
+    public EventHubClientBuilder proxyConfiguration(ProxyConfiguration proxyConfiguration) {
         this.proxyConfiguration = proxyConfiguration;
         return this;
     }
 
     /**
      * Sets the retry policy for EventHubClient.
      *
-     * @param retryPolicy The retry policy to use.
+     * @param retry The retry policy to use.
      * @return The updated EventHubClientBuilder object.
      */
-    public EventHubClientBuilder retry(RetryPolicy retryPolicy) {
-        this.retryPolicy = retryPolicy;
+    public EventHubClientBuilder retry(Retry retry) {
+        this.retry = retry;
+        return this;
+    }
+
+    /**
+     * Sets the configuration store that is used during construction of the service client.
+     *
+     * The default configuration store is a clone of the {@link ConfigurationManager#getConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     * @return The updated EventHubClientBuilder object.
+     */
+    public EventHubClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
         return this;
     }
 
     /**
      * Creates a new {@link EventHubClient} based on the configuration set in this builder.
+     * Use the default not null values if the Connection parameters are not provided.
      *
      * @return A new {@link EventHubClient} instance.
+     * @throws IllegalArgumentException when 'connectionString' is {@code null} or empty.
      * @throws AzureException If the token provider cannot be created for authorizing requests.
      */
     public EventHubClient build() {
+        this.configuration = this.configuration == null ? ConfigurationManager.getConfiguration().clone() : this.configuration;
```

Comment:
nit: You don't need to specify `this` for all your variable accesses in this method. There is no confusion between whether the variable belongs to the class or is scoped locally. imho, it adds extra reading when I already know we are trying to set the variables for the class. In the case of the constructor, where it's common to see the parameter variable the same as the member variable, then I'd have to use `this`.